docs/reference-guides/interactivity-api/iapi-about.md

```diff
@@ -0,0 +1,191 @@
+
+### Optional and gradual adoption
+
+The Interactivity API pipeline promotes **progressive enhancement** by building on top of WordPress’s solid foundation and patterns. It was carefully designed not to force any use cases to pay for the costs of other use cases.
+
+For example, blocks with directives can coexist with other (interactive or non-interactive) blocks. This means that if there are other blocks on the page using other frameworks like jQuery, everything will work as expected.
+
+### Declarative and reactive
+
+The Interactivity API follows an approach similar to other popular JS frameworks by separating state, actions, and callbacks and defining them declaratively. Why declaratively?
+
+Declarative code describes **what** a program should do, while imperative code describes **how** the program should do it. Using a declarative approach, the UI automatically updates in response to changes in the underlying data. With an imperative approach, you must manually update the UI whenever the data changes. Compare the two code examples:
+
+_Imperative code_
+
+```html
+<button id="toggle-button">Toggle Element</button>
+<p>This element is now visible!</p>
+<script>
+  const button = document.getElementById("toggle-button");
+ 
+  button.addEventListener("click", () => {
+    const element = document.getElementById("element");
+    if(element) {
+      element.remove();
+    } else {
+      const newElement = document.createElement("p");
+      newElement.textContent = "This element is visible";
+      document.body.appendChild(newElement);
+    }
+});
+</script>
+```
+
+_Declarative code_
+
+This is the same use case shared above but serves as an example of declarative code using this new system. The JavaScript logic is defined in the `view.js` file of the block, and add the directives to the markup in the `render.php`.
+
+```js
+// view.js file
+ 
+import { store, getContext } from "@wordpress/interactivity";
+ 
+store( 'wpmovies', {
+  actions: {
+    toggle: () => {
+      const context = getContext();
+      context.isOpen = !context.isOpen;
+    },
+  },
+});
+```
+
+```php
+<!-- Render.php file -->
+ 
+<div
+  data-wp-interactive='{ "namespace": "wpmovies" }'
+  data-wp-context="{ 'isOpen': true }"
+>
+  <button
+    data-wp-on--click="actions.toggle"
+    data-wp-bind--aria-expanded="context.ispen"
+    aria-controls="p-1"
+  >
+    Toggle
+  </button>
+ 
+  <p id="p-1" data-wp-show="context.isOpen">
+    This element is now visible!
+  </p>
+</div>
+```
+
+Using imperative code may be easier when creating simple user experiences, but it becomes much more difficult as blocks become more complex. The Interactivity API must cover all use cases, from the simplest to the most challenging. That’s why a declarative approach using directives better fits the Interactivity API.
```

Comment:
This is another chance to avoid being so block-centric:

```suggestion
Using imperative code may be easier when creating simple user experiences, but it becomes much more difficult as applications become more complex. The Interactivity API must cover all use cases, from the simplest to the most challenging. That’s why a declarative approach using directives better fits the Interactivity API.
```